Apromore-Custom-Plugins/Process-Discoverer-Portal-Plugin/src/main/java/org/apromore/plugin/portal/processdiscoverer/components/CaseDetailsController.java

```diff
@@ -78,6 +87,37 @@ public String getExportFilename() {
 		return parent.getContextData().getLogName() + ".csv";
 	}
 
+	/**
+	 * Update Activity <-> Attributes Map for quick lookup
+	 **/
+	private void updateActivityToAttributeMap(String caseId) {
+		activityToAttributeMap.clear();
```

Comment:
Can we move this method into some service class that already exists for eventlogs and usr tht service in this controller. 
We should avoid business logic here.

Reply:
Actually, there is no service associated with this component. And the map is UI construct, so it's more like a data transform into ViewModel rather than Model itself.

Reply:
Agree. This is the interaction on UI only. The interaction is like this: when user views the list of Cases, a window popup will show this list of cases in a table, each row is one case. When the user clicks on each row (a case), a linear visualization of the case is displayed in the form of a graph in PD (i.e. a straight line of boxes, each box is one event). Next, when the user hovers the mouse over one box (one event), there is a list of standard attributes of this event is shown in a tooltip box.